test/test_cpp_api_parity.py

```diff
@@ -180,19 +202,29 @@ def _compile_cpp_code_inline(self, name, cpp_sources, functions):
     def _test_torch_nn_module_ctor_args(self, module_name):
         python_module_class = getattr(torch.nn, module_name)
         module_metadata = torch_nn_modules.module_metadata_map[module_name]
-        cpp_default_constructor_args = module_metadata['cpp_default_constructor_args']
-
-        cpp_module_option = 'torch::nn::{}Options{}'.format(module_name, cpp_default_constructor_args)
+        cpp_default_constructor_args_str = module_metadata.cpp_default_constructor_args
         if PY2:
             init_arg_spec = inspect.getargspec(python_module_class.__init__)
         else:
             init_arg_spec = inspect.getfullargspec(python_module_class.__init__)
         init_kwargs_defaults = init_arg_spec.defaults
+        python_default_constructor_arg_names = [x for x in init_arg_spec.args[1:-len(init_kwargs_defaults)] if x != 'has_parity']
+        cpp_default_constructor_arg_values = cpp_default_constructor_args_str.strip('()').split(',')
+        assert len(cpp_default_constructor_arg_values) == len(python_default_constructor_arg_names), '''\
```

Comment:
There are a few ways to avoid the weird indentation. My favorite way is to use `self.assertEqual`:
```
self.assertEqual(len(...), len(...), 
   "your "
   "message "
   "here ")
```
and python should concatenate all of those lines.